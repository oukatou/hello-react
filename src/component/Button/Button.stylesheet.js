export default function stylesheet(props){
    return {
        lineHeight:'18px',
        fontSize:'0.8em',
        padding: '4px 10px',
        verticalAlign: 'middle',
        border: '1px solid transparent',
        borderRadius: '3px',
        outline: 'none',
        cursor: 'pointer',
        backgroundColor: '#afce31',
        color: '#fff',
        ...props.hasHover ? {
            boxShadow: '0 0 2px #afce31'
        } : {},
        ...props.isPressed ? {
            boxShadow: '0 0 0 2px #94b317'
        } : {}
    }
}