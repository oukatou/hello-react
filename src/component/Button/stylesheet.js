export default function stylesheet(props){
    return {
        display: 'inline-block',
        lineHeight:'18px',
        fontSize:'0.8em',
        padding: '4px 10px',
        verticalAlign: 'middle',
        border: '1px solid transparent',
        borderRadius: '3px',
        outline: 'none',
        cursor: 'pointer',
        backgroundColor: '#afce31',
        color: '#fff',
        width: props.width,
        ...props.disabled ? {
            opacity:0.5,
            cursor: 'default'} : {},
        ...props.hasHover ? {
            boxShadow: '0 0 2px #afce31'
        } : {},
        ...props.isPressed ? {
            boxShadow: '0 0 0 2px #94b317'
        } : {},
    }
}