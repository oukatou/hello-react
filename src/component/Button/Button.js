import React,{Component} from 'react'
import {css} from 'emotion'
import stylesheet from './stylesheet'
import ControlBehavior from '../Behavior/ControlBehavior'

export default class Button extends Component{
    render(){
        let {
            title,
            onClick,
            onMouseDown,
            onMouseUp,
            onMouseEnter,
            onMouseLeave,
            disabled,
            width
        } = this.props
        
        return (
            <ControlBehavior
                onMouseDown={onMouseDown}
                onMouseLeave={onMouseLeave}
                onMouseUp={onMouseUp}
                onMouseEnter={onMouseEnter}
            >
                {
                    ({  hasHover,
                        isPressed,
                        onMouseDown: handleMouseDown,
                        onMouseEnter: handleMouseEnter,
                        onMouseLeave: handleMouseLeave,
                        onMouseUp: handleMouseUp
                    })=>
                        <button onClick={onClick} 
                                onMouseDown={handleMouseDown}
                                onMouseLeave={handleMouseLeave}
                                onMouseUp={handleMouseUp}
                                onMouseEnter={handleMouseEnter}
                                className={css(stylesheet({width, disabled,isPressed,hasHover}))}>
                            <span>{title}</span>
                        </button>
                }
            </ControlBehavior>
        )
    }
}
