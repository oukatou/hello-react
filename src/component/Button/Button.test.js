/* eslint-disable no-undef */
import React from 'react'
import {mount} from 'enzyme'
import Button from './Button'
import renderer from 'react-test-renderer';

describe('<Button/>',()=>{
    let wrapper,eventHandler
    let subject = (props) => mount(<Button {...props}></Button>)
    describe("onClick", () => {
        beforeEach(() => {
          eventHandler = jest.fn();
          wrapper = subject({ onClick: eventHandler });
        });
  
        it("is triggered on click", () => {
          wrapper.simulate("click");
          expect(eventHandler).toBeCalled();
        });
      });
    
    describe("onMouseLeave", () => {
        beforeEach(() => {
          eventHandler = jest.fn();
          wrapper = subject({ onMouseLeave: eventHandler });
        });
  
        it("is triggered on MouseLeave", () => {
          wrapper.simulate("mouseleave");
          expect(eventHandler).toBeCalled();
        });
      });

    describe("onMouseEnter", () => {
        beforeEach(() => {
          eventHandler = jest.fn();
          wrapper = subject({ onMouseEnter: eventHandler });
        });
  
        it("is triggered on MouseEnter", () => {
          wrapper.simulate("mouseenter");
          expect(eventHandler).toBeCalled();
        });
      });
    /*describe("<button/>", () => {
        beforeEach(() => {
          wrapper = subject({ title: "test" }  );
        });
        it("is a <button> tag", () => {
            expect(wrapper.matchesElement(<button><span>test</span></button>)).toEqual(true);
          });
      }); */
    const props = { title: "test" }  
    const vDom = React.createElement(Button, props);
    const tree = renderer.create(vDom).toJSON();
    expect(tree).toMatchSnapshot()

})