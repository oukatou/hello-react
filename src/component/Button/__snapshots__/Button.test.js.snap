// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
.emotion-0 {
  line-height: 18px;
  font-size: 0.8em;
  padding: 4px 10px;
  vertical-align: middle;
  border: 1px solid transparent;
  border-radius: 3px;
  outline: none;
  cursor: pointer;
  background-color: #afce31;
  color: #fff;
}

<button
  className="emotion-0"
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
>
  <span>
    test
  </span>
</button>
`;
